app_test.py

```
@@ -118,7 +118,22 @@ def user_input_images(query):
         # Hiển thị từng ảnh trong danh sách
         for image in list_images:
             st.image(image, caption=image, use_container_width=True)
+
 if __name__ == "__main__":
-    if query_text_input != '':
-        user_input(query_text_input)
-        user_input_images(query_text_input)
\ No newline at end of file
+    try:
+        query_text_input = input("Nhập câu truy vấn: ").strip()
+
+        if query_text_input:
+            print(f"✅ Input received: {query_text_input}")
+
+            # Gọi xử lý text
+            user_input(query_text_input)
+
+            # Gọi xử lý ảnh
+            user_input_images(query_text_input)
+
+        else:
+            print("⚠️ Không có nội dung được nhập!")
+
```

Avoid catching Exception broadly. Use more specific exception types where possible. If you must catch all exceptions, at least log the full traceback (e.g., using logging.exception) so debugging is easier, rather than printing str(e).